component/src/governance/execute.rs

```diff
@@ -126,6 +111,40 @@ pub async fn validator_vote(
     Ok(())
 }
 
+#[instrument(skip(state))]
+pub async fn proposal_deposit_claim(
+    state: &mut StateTransaction<'_>,
+    ProposalDepositClaim {
+        proposal,
+        deposit_amount: _, // not needed to transition state; deposit is self-minted in tx
+        outcome: resupplied_outcome,
+    }: &ProposalDepositClaim,
+) -> Result<()> {
+    // The only effect of doing a deposit claim is to state transition the proposal to claimed so it
+    // cannot be claimed again. The deposit amount is self-minted in the transaction (proof of
+    // deserving-ness is the supplied proposal NFT, which is burned in the transaction), so we don't
+    // need to distribute it here.
+
+    if let Some(proposal::State::Finished { outcome }) = state.proposal_state(*proposal).await? {
+        // This should be prevented by earlier checks, but replicating here JUST IN CASE!
+        if *resupplied_outcome != outcome.as_ref().map(|_| ()) {
+            anyhow::bail!(
+                "proposal {} has outcome {:?}, but deposit claim has outcome {:?}",
+                proposal,
+                outcome,
+                resupplied_outcome
+            );
+        }
```

Comment:
Should we recheck this here? Should we also recheck the proposal deposit amount matches?